From 2c3ba09e648051444757850af25570ae55c38189 Mon Sep 17 00:00:00 2001
From: Soren Brinkmann <soren.brinkmann@xilinx.com>
Date: Wed, 26 Aug 2015 11:56:37 -0700
Subject: [PATCH 072/827] zynqmp: PM: Add clobber list for inline assembly

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

The SMC calling conventions allow firmware to modify x0-x17. Add those
registers to the clobber list.

Signed-off-by: Soren Brinkmann <soren.brinkmann@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit ef5a49bc14b7bb08b19bc794b31c8a48ff6c1f3c)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/soc/xilinx/zynqmp/pm.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/soc/xilinx/zynqmp/pm.c b/drivers/soc/xilinx/zynqmp/pm.c
index a7a35c9..947bc15 100644
--- a/drivers/soc/xilinx/zynqmp/pm.c
+++ b/drivers/soc/xilinx/zynqmp/pm.c
@@ -133,6 +133,9 @@ static noinline int do_fw_call_smc(u64 arg0, u64 arg1, u64 arg2,
 		__asmeq("%2", "x2")
 		"smc	#0\n"
 		: "+r" (arg0), "+r" (arg1), "+r" (arg2)
+		: /* no input only */
+		: "x3", "x4", "x5", "x6", "x7", "x8", "x9", "x10", "x11", "x12",
+		  "x13", "x14", "x15", "x16", "x17"
 		);
 
 	if (ret_payload != NULL) {
@@ -172,6 +175,9 @@ static noinline int do_fw_call_hvc(u64 arg0, u64 arg1, u64 arg2,
 		__asmeq("%2", "x2")
 		"hvc	#0\n"
 		: "+r" (arg0), "+r" (arg1), "+r" (arg2)
+		: /* no input only */
+		: "x3", "x4", "x5", "x6", "x7", "x8", "x9", "x10", "x11", "x12",
+		  "x13", "x14", "x15", "x16", "x17"
 		);
 
 	if (ret_payload != NULL) {
-- 
2.9.3

